lib/FrontendTool/ReferenceDependencies.cpp

```diff
@@ -123,153 +137,155 @@ swift::reversePathSortedFilenames(const ArrayRef<std::string> elts) {
+  auto *NTD = ED->getExtendedType()->getAnyNominal();
+  if (!NTD || !isVisibleOutside(NTD))
+    return;
+
+  // Check if the extension is just adding members, or if it is
+  // introducing a conformance to a public protocol.
+  bool justMembers =
+      std::all_of(ED->getInherited().begin(),
+                  ED->getInherited().end(),
+                  extendedTypeIsPrivate);
+  if (justMembers) {
+    if (std::all_of(ED->getMembers().begin(), ED->getMembers().end(),
+                    declIsPrivate))
+      return;
+    extensionsWithJustMembers.push_back(ED);
   }
+  extendedNominals[NTD] |= !justMembers;
+  findNominalsAndOperators(extendedNominals, memberOperatorDecls,
+                           ED->getMembers());
+}
 
-  auto escape = [](DeclBaseName name) -> std::string {
-    return llvm::yaml::escape(name.userFacingName());
-  };
-
-  out << "### Swift dependencies file v0 ###\n";
-
-  llvm::MapVector<const NominalTypeDecl *, bool> extendedNominals;
-  llvm::SmallVector<const FuncDecl *, 8> memberOperatorDecls;
-  llvm::SmallVector<const ExtensionDecl *, 8> extensionsWithJustMembers;
-
-  out << "provides-top-level:\n";
-  for (const Decl *D : SF->Decls) {
-    switch (D->getKind()) {
-    case DeclKind::Module:
-      break;
-
-    case DeclKind::Import:
-      // FIXME: Handle re-exported decls.
-      break;
-
-    case DeclKind::Extension: {
-      auto *ED = cast<ExtensionDecl>(D);
-      auto *NTD = ED->getExtendedType()->getAnyNominal();
-      if (!NTD)
-        break;
-      if (NTD->hasAccess() &&
-          NTD->getFormalAccess() <= AccessLevel::FilePrivate) {
-        break;
-      }
+static void writeNominalTypeDecl(
+    const NominalTypeDecl *NTD, llvm::raw_fd_ostream &out,
+    llvm::MapVector<const NominalTypeDecl *, bool> &extendedNominals,
+    llvm::SmallVectorImpl<const FuncDecl *> &memberOperatorDecls) {
+  if (!hasVisibleName(NTD))
+    return;
+
+  out << "- \"" << escape(NTD->getName()) << "\"\n";
+  extendedNominals[NTD] |= true;
+  findNominalsAndOperators(extendedNominals, memberOperatorDecls,
+                           NTD->getMembers());
+}
 
-      // Check if the extension is just adding members, or if it is
-      // introducing a conformance to a public protocol.
-      bool justMembers = std::all_of(ED->getInherited().begin(),
-                                     ED->getInherited().end(),
-                                     extendedTypeIsPrivate);
-      if (justMembers) {
-        if (std::all_of(ED->getMembers().begin(), ED->getMembers().end(),
-                        declIsPrivate)) {
-          break;
-        } else {
-          extensionsWithJustMembers.push_back(ED);
-        }
-      }
-      extendedNominals[NTD] |= !justMembers;
-      findNominalsAndOperators(extendedNominals, memberOperatorDecls,
-                               ED->getMembers());
-      break;
-    }
+static void writeValueDecl(const ValueDecl *VD, llvm::raw_fd_ostream &out) {
```

Comment:
Nominals are also value decls, so it's not clear why you'd call one and not the other. (This is an example where breaking things up into multiple functions loses context!) In practice the tidiest answer may be for writeNominalTypeDecl to chain to writeValueDecl.

Reply:
I'm confused (because I don't know the Decl types well):
`writeNominalTypeDecl` uses `getName`, but `writeValueDecl` uses `getBaseName`.  If one chains to the other, should both use the same one? Which?
TIA.

Reply:
Nominal types never have compound names, so `getName` and `getBaseName` will produce the same effective result. It's fine to use `getBaseName` for all of them.

Reply:
OK, will do. But it's sorta a change in functionality.

Reply:
When you write that "nominals are also value decls", I don't know quite what you mean in terms I can code to. If I try to chain `writeNominalTypeDecl` to `writeValueTypeDecl`, there is a type error because a `NominalTypeDecl` is not a subtype of a `ValueDecl`. I *could* bypass the type system with a template in the style of `hasSomePrivateAccessLevel`, but I don't think that's what you mean.